src/components/ellenTicTacToe.tsx

```diff
@@ -0,0 +1,240 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { css } from 'emotion';
+function Square(props: {
+  value: string;
+  onClick: () => void;
+  isWinning: string;
+}) {
+  return (
+    <button
+      className={css(
+        {
+          backgroundColor:
+            props.isWinning === 'winning square' ? 'lightgreen' : 'white',
+          border: '1px solid #999',
+          float: 'left',
+          font: 'inherit',
+          fontSize: '30px',
+          fontWeight: 'bold',
+          height: '50px',
+          lineHeight: '34px',
+          margin: 0,
+          marginRight: '-1px',
+          marginTop: '-1px',
+          overflow: 'visible',
+          padding: 0,
+          textAlign: 'center',
+          textTransform: 'none',
+          width: '50px',
+        },
+        { ':focus': { outline: 'none' } },
+      )}
+      onClick={props.onClick}
+    >
+      {props.value}
+    </button>
+  );
+}
+interface BoardProps {
+  squares: string[];
+  onClick: (i: number) => void;
+  winningArray: string[];
+}
+class Board extends React.Component<BoardProps, {}> {
+  renderSquare(i: number) {
+    return (
+      <Square
+        value={this.props.squares[i]}
+        onClick={() => this.props.onClick(i)}
+        isWinning={this.props.winningArray[i]}
+      />
+    );
+  }
+  render() {
+    return (
+      <div>
+        <div className="board-row">
+          {this.renderSquare(0)}
+          {this.renderSquare(1)}
+          {this.renderSquare(2)}
+        </div>
+        <div className="board-row">
+          {this.renderSquare(3)}
+          {this.renderSquare(4)}
+          {this.renderSquare(5)}
+        </div>
+        <div className="board-row">
+          {this.renderSquare(6)}
+          {this.renderSquare(7)}
+          {this.renderSquare(8)}
+        </div>
+      </div>
+    );
+  }
+}
+interface GameState {
+  history: Array<{ squares: string[]; clickedSquare?: number[] }>;
+  stepNumber: number;
+  xIsNext: boolean;
+}
+// tslint:disable-next-line:max-classes-per-file
+export class Game extends React.Component<{}, GameState> {
+  constructor(props: {}) {
+    super(props);
+    this.state = {
+      history: [
+        {
+          squares: Array(9).fill(null),
+        },
+      ],
+      stepNumber: 0,
+      xIsNext: true,
+    };
+  }
+  handleClick(i: number) {
+    const history = this.state.history.slice(0, this.state.stepNumber + 1);
+    const current = history[history.length - 1];
+    const squares = current.squares.slice();
+    if (calculateWinner(squares) || squares[i]) {
+      return;
+    }
+    squares[i] = this.state.xIsNext ? 'X' : 'O';
+    this.setState({
+      history: history.concat([
+        {
+          clickedSquare: [Math.floor((i % 3) + 1), Math.floor(i / 3 + 1)],
+          squares,
+        },
+      ]),
+      stepNumber: history.length,

```

Comment:
You don't need `stepNumber`; it's just a pure function of `history`. If you want to save writing out `this.state.history.length` because it's used quite often, it'd be better to define a method `stepNumber() { return this.state.history.length }`.